graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/conversion/IsBoolean.java

```diff
@@ -23,29 +23,43 @@
 import org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor;
 import org.graylog.plugins.pipelineprocessor.rulebuilder.RuleBuilderFunctionGroup;
 
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.bool;
 import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.object;
 
 public class IsBoolean extends AbstractFunction<Boolean> {
     public static final String NAME = "is_bool";
 
     private final ParameterDescriptor<Object, Object> valueParam;
+    private final ParameterDescriptor<Boolean, Boolean> conversionParam;
 
     public IsBoolean() {
         valueParam = object("value").ruleBuilderVariable().description("Value to check").build();
+        conversionParam = bool("attemptConversion").optional().description("Try to convert value to boolean from its string representation (default: false)").build();
     }
 
     @Override
     public Boolean evaluate(FunctionArgs args, EvaluationContext context) {
         final Object value = valueParam.required(args, context);
-        return value instanceof Boolean;
+        final boolean convert = conversionParam.optional(args, context).orElse(false);
+        if (!convert) {
+            return value instanceof Boolean;
+        }
+        if (value instanceof Boolean) {
```

Comment:
would check here with BooleanUtils.toBooleanObject(value) instanceof Boolean and discard the parsing as for false it would other wise return false but that is still a boolean

Reply:
thanks for spotting this